### PR TITLE
♻️ (tooltip) refactoring to react-aria (vol.4)

### PR DESCRIPTION
**Pull request by MatissJanis**
Follow-up to https://github.com/actualbudget/actual/pull/2631

This should move the majority (if not all) of the Tooltip components used in the reports page.

### CODE REVIEW

File: packages/desktop-client/src/components/budget/BudgetTotals.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import React, { type ComponentProps, memo, useState } from 'react';
+import React, { type ComponentProps, memo, useRef, useState } from 'react';
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/BudgetTotals.tsx:1]: Recommendation when reviewing:
Screenshot 2024-05-07 at 17 28 22

File: packages/desktop-client/src/components/budget/rollover/CoverTooltip.tsx [hunk 2]
```diff
@@ -1,46 +1,18 @@
...
-    </Tooltip>
-  );
-}
-
-type CoverProps = {
-  onSubmit: (categoryId: string) => void;
-};
-
-function Cover({ onSubmit }: CoverProps) {
+export function CoverTooltip({ onSubmit, onClose }: CoverTooltipProps) {
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/CoverTooltip.tsx:15]: We should consider renaming this as they're no longer tooltips. Maybe `CoverMenu`?

File: packages/desktop-client/src/components/budget/rollover/TransferTooltip.tsx [hunk 3]
```diff
@@ -10,52 +9,21 @@ import { Button } from '../../common/Button';
...
 
-type TransferTooltipProps = ComponentPropsWithoutRef<typeof Tooltip> & {
+type TransferTooltipProps = {
   initialAmount?: number;
   showToBeBudgeted?: boolean;
   onSubmit: (amount: number, categoryId: string) => void;
+  onClose: () => void;
 };
 
 export function TransferTooltip({
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/TransferTooltip.tsx:21]: We should consider renaming this as they're no longer tooltips. Maybe `TransferMenu`?

File: packages/desktop-client/src/components/budget/rollover/HoldTooltip.tsx [hunk 4]
```diff
@@ -15,17 +14,12 @@ import { InitialFocus } from '../../common/InitialFocus';
...
   onSubmit: (amount: number) => void;
+  onClose: () => void;
 };
-export function HoldTooltip({
-  onSubmit,
-  onClose,
-  position = 'bottom-right',
-  ...props
-}: HoldTooltipProps) {
+export function HoldTooltip({ onSubmit, onClose }: HoldTooltipProps) {
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/HoldTooltip.tsx:22]: We should consider renaming this as it's no longer a tooltip. Maybe `HoldMenu`?

File: packages/desktop-client/src/components/budget/rollover/BalanceTooltip.tsx [hunk 5]
```diff
@@ -3,66 +3,49 @@ import React, { useState } from 'react';
...
 
 type BalanceTooltipProps = {
   categoryId: string;
-  tooltip: { close: () => void };
   month: string;
   onBudgetAction: (month: string, action: string, arg?: unknown) => void;
   onClose?: () => void;
 };
 
 export function BalanceTooltip({
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/BalanceTooltip.tsx:18]: Same comment as other tooltips

File: packages/desktop-client/src/components/budget/rollover/CoverTooltip.tsx (hunk 2 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/budget/rollover/CoverTooltip.tsx]: Didn't want to do too much changes in this PR, but that makes sense :) 
Updated

File: packages/desktop-client/src/components/budget/rollover/BalancePopoverContent.tsx [hunk 6]
```diff
@@ -3,66 +3,49 @@ import React, { useState } from 'react';
...
+type BalancePopoverContentProps = {
   categoryId: string;
-  tooltip: { close: () => void };
   month: string;
   onBudgetAction: (month: string, action: string, arg?: unknown) => void;
   onClose?: () => void;
 };
 
-export function BalanceTooltip({
+export function BalancePopoverContent({
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/rollover/BalancePopoverContent.tsx:18]: I would suggest `BalanceMovementMenu` here as we could potentially use this out of the context of popovers.

Comment (MatissJanis) [reply packages/desktop-client/src/components/budget/rollover/BalancePopoverContent.tsx]: Good idea! https://github.com/actualbudget/actual/pull/2724/commits/d0291ac175fbee88627a05f1ad6aebcd5251680f

**Comment by joel-jeremy** [comment]
The category / group popover is in the wrong position

**Comment by MatissJanis** [comment]
> The category / group popover is in the wrong position

Which one is it? Can you send a screenshot?

**Comment by joel-jeremy** [comment]
@MatissJanis The popover appears on the top left part of the page
image

**Comment by MatissJanis** [comment]
@joel-jeremy is it a specific category/group menu that appears like that or is it them all?

This is what I have:

https://github.com/actualbudget/actual/assets/886567/89218330-ffc6-48ee-8502-c05eec05f173

**Comment by MatissJanis** [comment]
Managed to reproduce.

Safari is OK. Chrome breaks.

**Comment by MatissJanis** [comment]
Thanks for testing so carefully @joel-jeremy ! 🙇

**Review by joel-jeremy — APPROVED** [review]
LGTM!

Just curious - why we chose different conventions for the `Tooltip` and `Popover` components.  Popover uses triggerRef but Tooltip wraps the trigger element. Would it be possible to use a single pattern for both? Just thought to bring it up, but that's out of scope of this PR.

**Comment by MatissJanis** [comment]
> Just curious - why we chose different conventions for the `Tooltip` and `Popover` components. Popover uses triggerRef but Tooltip wraps the trigger element. Would it be possible to use a single pattern for both? Just thought to bring it up, but that's out of scope of this PR.

Ideally I would have liked them both to just wrap the trigger element (instead of using refs), but that's currently not possible. We'd need to port buttons too to `react-aria` to remove the `triggerRef` from `Popover`.

**Pull request merged**